agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,755 @@
+        if not task_uuid:
+            logging.error("Task UUID was not provided")
+            raise AHVFenceAgentException("Task UUID not provided")
+
+        task_url = f"{self.task_url}/{task_uuid}"
+        header_str = self._get_headers()
+        task_resp = None
+        interval = 10
+        task_status = None
+
+        if not timeout:
+            timeout = MIN_TIMEOUT
+
+        while task_status != 'SUCCEEDED':
+            if task_status == 'FAILED':
+                raise NutanixClientException(f"Task failed, task uuid: {task_uuid}")
+
+            try:
+                task_resp = self.request(url=task_url, method='GET',
+                                         headers=header_str, verify=self.verify)
+                task_status = task_resp.json()['data']['status']
+            except NutanixClientException as err:
+                logging.error("Unable to retrieve task status")
+                raise AHVFenceAgentException from err
+
+            if task_status == 'SUCCEEDED':
+                break
+
+            time.sleep(interval)
+            timeout = timeout - interval
+
+            if timeout <= 0:
+                raise AHVFenceAgentException("Timed out waiting"
+                                             f" for task: {task_uuid}")
+
+    def list_vms(self, filter_str=None, limit=None):
+        """
+        List all VM's name, UUID, and power state
+
+        Args:
+            filter_str(str): filter string.
+
+        Returns:
+            list: A list of all VMs (Name, UUID, power state).
+        """
+        vms = self._get_all_vms(filter_str, limit)
+        vm_list = []
+
+        for vm in vms['data']:
+            vm_name = vm['name']
+            ext_id = vm['extId']
+            power_state = vm['powerState']
+            vm_data = f"{vm_name},{ext_id},{power_state}"
+            vm_list.append(vm_data)
+
+        return vm_list
+
+    def get_power_state(self, vm_name=None, vm_uuid=None):
+        """
+        Get power state of a VM.
+
+        Args:
+            task_uuid(str): Task UUID.
+
+        Returns:
+            str: A json formatted response to the HTTP request.
+        """
+        resp = None
+
+        if not vm_name and not vm_uuid:
+            logging.error("Require at least one of VM name or VM UUID")
+            raise AHVFenceAgentException("No arguments provided")
+
+        if not vm_uuid:
+            try:
+                vm_uuid = self._get_vm_uuid(vm_name)
+            except AHVFenceAgentException as err:
+                logging.error("Unable to retrieve UUID of VM, %s", vm_name)
+                raise AHVFenceAgentException from err
+
+        try:
+            resp = self._get_vm(vm_uuid)
+        except AHVFenceAgentException as err:
+            logging.error("Unable to retrieve power state of VM %s", vm_uuid)
+            raise AHVFenceAgentException from err
+
+        power_state = resp.json()['data']['powerState']
+
+        if power_state.lower() != 'on':
```

Comment:
we could have a pause state! not common but if acropolis is down I believe the VM can be in paused state!

Reply:
Yes, I agree. In this case I am only interested if the VM is powered-on or not.